TestResetPasswordHandler-2 fails on go 1.4.3

Seen in #271 (a documentation change).

Have seen this before, but rerunning the tests on Travis fixed it. Feels like a test that might only fail every once in a while.

```
--- FAIL: TestResetPasswordHandler-2 (1.13s)
	password_test.go:567: case 0: Compare(wantFormValues, got) =  {
		  password: [
		             "",
		            ],
		  token:    [
		-            "eyJhbGciOiJSUzI1NiIsImtpZCI6InRzMjJlQUZNTlE1REwycWpXNHpPbUNuTjZ2VHdDMDMyc1J6d3p2SjhIa0pSb3IzeFF1cnZ3amlFcE5jajhQQ0s5eTlzOEhtaERNYVVnNUZoZjBPZkdWdG5XSXFVdng3Q0ZORWU0LTZwZnNlemZwVkFpbE9aQjc0alR1YmZDOExXMnVMUXFZemFWeGlWQ1I2VDA5ZzdBdDNYb2h6bFBtdE54X210dWFnd2hwMWR2UzNsMVUyVGZfcXBwejdsX0pXTTJTSEhJWDhSOGl0ZEI0Y3VxbmtHY0pZMTZOUVJkU0lJcDFDTnhlbTNONkZPQlRpZW1mVG0xdG96VUxIMEx4emNjcGVEQkhNdlRMaTZ1eFRJTU1UbGVqUDNVbXZtbFNoY21rT0VvZmMwNm5Kd2dXOVVfX09GanFIY1hoanFhSUhUX3pFdTdOOExDbGtKZVRsN1lYc3J6dz09IiwidHlwIjoiSldUIn0.eyJhdWQiOiJYWFgiLCJleHAiOjEuNDUyODg3NjE1ZSswOSwiaHR0cDovL2NvcmVvcy5jb20vcGFzc3dvcmQvb2xkLWhhc2giOiJwYXNzd29yZCIsImh0dHA6Ly9jb3Jlb3MuY29tL3Bhc3N3b3JkL3Jlc2V0LWNhbGxiYWNrIjoiaHR0cDovL2NsaWVudC5leGFtcGxlLmNvbS9jYWxsYmFjayIsImlhdCI6MS40NTI4ODQwMTVlKzA5LCJpc3MiOiJodHRwOi8vc2VydmVyLmV4YW1wbGUuY29tIiwic3ViIjoiSUQtMSJ9.gKjJKJ6urdvI4md93btv5Nt3TUX6rEAboqPmTWVeyF2F_9ipmq2W87ortJLyRebQ1zndU3tq4Iud_lyn8e4W7EDTRYgMKWstoFyTKYyzhdnOHUL85Fsx8NvT8QinCj9RMTna2W1ovhJe2oqpjIN6J4GQWh3Bg2D6mDHnxowd2oHexVhSddMqqGjAzUBKB2jUpodzx2TjUka5JS94G9UXcH9fkayZSRe5A-2LxxIEo8WrXbyAOxvzjkxwVQIEcG9OnjKnpz1wWpQ1BptaY08gHDzQ8vRvB19y59zp4tNp2bzq6pRCntK0065g0QQoxK9W_pbQlWO4K7Xf2iBrJKzDSg",
		+            "eyJhbGciOiJSUzI1NiIsImtpZCI6InRzMjJlQUZNTlE1REwycWpXNHpPbUNuTjZ2VHdDMDMyc1J6d3p2SjhIa0pSb3IzeFF1cnZ3amlFcE5jajhQQ0s5eTlzOEhtaERNYVVnNUZoZjBPZkdWdG5XSXFVdng3Q0ZORWU0LTZwZnNlemZwVkFpbE9aQjc0alR1YmZDOExXMnVMUXFZemFWeGlWQ1I2VDA5ZzdBdDNYb2h6bFBtdE54X210dWFnd2hwMWR2UzNsMVUyVGZfcXBwejdsX0pXTTJTSEhJWDhSOGl0ZEI0Y3VxbmtHY0pZMTZOUVJkU0lJcDFDTnhlbTNONkZPQlRpZW1mVG0xdG96VUxIMEx4emNjcGVEQkhNdlRMaTZ1eFRJTU1UbGVqUDNVbXZtbFNoY21rT0VvZmMwNm5Kd2dXOVVfX09GanFIY1hoanFhSUhUX3pFdTdOOExDbGtKZVRsN1lYc3J6dz09IiwidHlwIjoiSldUIn0.eyJhdWQiOiJYWFgiLCJleHAiOjEuNDUyODg3NjE0ZSswOSwiaHR0cDovL2NvcmVvcy5jb20vcGFzc3dvcmQvb2xkLWhhc2giOiJwYXNzd29yZCIsImh0dHA6Ly9jb3Jlb3MuY29tL3Bhc3N3b3JkL3Jlc2V0LWNhbGxiYWNrIjoiaHR0cDovL2NsaWVudC5leGFtcGxlLmNvbS9jYWxsYmFjayIsImlhdCI6MS40NTI4ODQwMTRlKzA5LCJpc3MiOiJodHRwOi8vc2VydmVyLmV4YW1wbGUuY29tIiwic3ViIjoiSUQtMSJ9.ceaP7Bdq1ZxzpIR3Vx-Robl6pF7NeL3thYzykS54G8q_BOHHl67FheF6xJXyIaF_WD3TYVNv2lQwF8HF2rW01x-8v9oiZikMo4v8fKyURxQ-UUg5-kQIX5oB6LbBl4kE2vFtxOBUzKcTdX2FpqXfrdZ7A2-Hz2-skWUyJq7CnoEzEoTpdCAfzU0PazhtYRlF9loHYSnhWa5lIOgbvDWapANYJb216Q6jXuu9OjlgZ0D5l4Zhwr4CrrsdjN-wD4GIPPSZz5koXUv76VhCKxh4FjdRPOfY8Z9VhK--D3cV7tXvfMWp0hI5tJRV--EyTateNsB0Q8pvv0kb3ha8Spkqwg",
		            ],
		 }
	password_test.go:568: case 0: Body: <!DOCTYPE html>
```

Full logs here: https://gist.github.com/ericchiang/38bffbc7b389f52df543

--- a/server/password_test.go
+++ b/server/password_test.go
@@ -392,6 +392,12 @@ func TestResetPasswordHandler(t *testing.T) {
 		user.PasswordHasher = user.DefaultPasswordHasher
 	}()
 
+	tokenForCase := map[int]string{
+		0: makeToken("ID-1", "password", testClientID, testRedirectURL, time.Hour*1, goodSigner),
+		2: makeToken("ID-1", "password", testClientID, url.URL{}, time.Hour*1, goodSigner),
+		5: makeToken("ID-1", "password", testClientID, url.URL{}, time.Hour*1, goodSigner),
+	}
+
 	tests := []struct {
 		query url.Values
 
@@ -405,14 +411,14 @@ func TestResetPasswordHandler(t *testing.T) {
 		{ // Case 0
 			// Step 1.1 - User clicks link in email, has valid token.
 			query: url.Values{
-				"token": str(makeToken("ID-1", "password", testClientID, testRedirectURL, time.Hour*1, goodSigner)),
+				"token": str(tokenForCase[0]),
 			},
 			method: "GET",
 
 			wantCode: http.StatusOK,
 			wantFormValues: &url.Values{
 				"password": str(""),
-				"token":    str(makeToken("ID-1", "password", testClientID, testRedirectURL, time.Hour*1, goodSigner)),
+				"token":    str(tokenForCase[0]),
 			},
 			wantPassword: "password",
 		},
@@ -432,14 +438,14 @@ func TestResetPasswordHandler(t *testing.T) {
 		{ // Case 2
 			// Step 2.1 - User clicks link in email, has valid token.
 			query: url.Values{
-				"token": str(makeToken("ID-1", "password", testClientID, url.URL{}, time.Hour*1, goodSigner)),
+				"token": str(tokenForCase[2]),
 			},
 			method: "GET",
 
 			wantCode: http.StatusOK,
 			wantFormValues: &url.Values{
 				"password": str(""),
-				"token":    str(makeToken("ID-1", "password", testClientID, url.URL{}, time.Hour*1, goodSigner)),
+				"token":    str(tokenForCase[2]),
 			},
 			wantPassword: "password",
 		},
@@ -472,7 +478,7 @@ func TestResetPasswordHandler(t *testing.T) {
 		{ // Case 5
 			// Step 2.2.1 - User enters in new valid password, password is changed, no redirect
 			query: url.Values{
-				"token":    str(makeToken("ID-1", "password", testClientID, url.URL{}, time.Hour*1, goodSigner)),
+				"token":    str(tokenForCase[5]),
 				"password": str("shrt"),
 			},
 			method: "POST",
@@ -481,7 +487,7 @@ func TestResetPasswordHandler(t *testing.T) {
 			wantCode: http.StatusBadRequest,
 			wantFormValues: &url.Values{
 				"password": str(""),
-				"token":    str(makeToken("ID-1", "password", testClientID, url.URL{}, time.Hour*1, goodSigner)),
+				"token":    str(tokenForCase[5]),
 			},
 			wantPassword: "password",
 		},
@@ -553,7 +559,6 @@ func TestResetPasswordHandler(t *testing.T) {
 
 		if tt.wantCode != w.Code {
 			t.Errorf("case %d: wantCode=%v, got=%v", i, tt.wantCode, w.Code)
-			t.Logf("case %d: Body: %v ", i, w.Body)
 			continue
 		}
 
@@ -565,7 +570,6 @@ func TestResetPasswordHandler(t *testing.T) {
 		if tt.wantFormValues != nil {
 			if diff := pretty.Compare(*tt.wantFormValues, values); diff != "" {
 				t.Errorf("case %d: Compare(wantFormValues, got) = %v", i, diff)
-				t.Logf("case %d: Body: %v ", i, w.Body)
 			}
 		}
 		pwi, err := f.srv.PasswordInfoRepo.Get(nil, "ID-1")

